v1/membership_update_message_test.go

```diff
@@ -48,3 +49,22 @@ func TestEncodeMembershipUpdateMessage(t *testing.T) {
 	require.NoError(t, err)
 	require.EqualValues(t, testMembershipUpdateMessageStruct, val)
 }
+
+func TestSignMembershipUpdate(t *testing.T) {
+	event := testMembershipUpdateMessageStruct.Updates[0].Events[0]
+	result := tupleMembershipUpdateEvent(event)
+	require.EqualValues(t, [][]interface{}{
+		{"clock-value", event.ClockValue},
+		{"name", "thathata"},
+		{"type", "chat-created"},
+	}, result)
+
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	update := testMembershipUpdateMessageStruct.Updates[0]
+	err = SignMembershipUpdate(&update, key)
+	require.NoError(t, err)
+
+	// TODO: we should generate a signature in ClojureScript and
```

Comment:
@cammellos would it be easy to generate? I am thinking of sharing a private key, signature and events between this lib and status-react. This way we can ensure in unit tests that both implementations are correct and prevent future incompatibilities.

Reply:
Not sure I understand exactly, I can generate probably a signature for a given private key, but:

Currently the signing happens in `status-go`: https://github.com/status-im/status-go/blob/develop/mobile/status.go#L88

we pass a JSON formatted string from `status-react` to `status-go` that will be signed:
`"[[[....]]]'`

So I would have only one part of the codebase to generate that (`status-protocol-go`),
to increase the surface of shared code (so that we slowly move to use status-protocol-go for private group-chats), I would pass instead of a string a golang struct, `MembershipUpdate` (which should be compatible), and just have one place we do the signing.

This way we have a single place, so `status-react` and `status-protocol-go` won't have to be kept synchronized in the way the sign messages (only `status-protocol-go` will do.). Also no need to make sure the signature is identical with the one generated by status-react/status-go (although they should be, and if not we should understand why they are not), as we don't need to keep backward compatibility.

What do you think? did I understand what you mean?

Reply:
That totally makes sense. I was worried about the serialization to a string `[[[...]]]` and keeping compatibility with the current codebase. But, as you pointed out that backward compatibility is not an issue, we should concentrate all codebase in this lib and it won't be a problem.

No action needed :) Thanks!